src/Benchmarks/Data/ApplicationDbContext.cs

```diff
@@ -22,7 +27,34 @@ public ApplicationDbContext(IOptions<AppSettings> appSettings)
 
         protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         {
-            optionsBuilder.UseSqlServer(_appSettings.ConnectionString);
+            var extension = GetOrCreateExtension(optionsBuilder);
+            extension.ConnectionString = _appSettings.ConnectionString;
+            ((IDbContextOptionsBuilderInfrastructure)optionsBuilder).AddOrUpdateExtension(extension);
+        }
+
+        private static SqlServerOptionsExtension GetOrCreateExtension(DbContextOptionsBuilder optionsBuilder)
+        {
+            var existing = optionsBuilder.Options.FindExtension<NoTxSqlServerOptionsExtension>();
+            return existing != null
+                ? new NoTxSqlServerOptionsExtension(existing)
+                : new NoTxSqlServerOptionsExtension();
+        }
+
+        private class NoTxSqlServerOptionsExtension : SqlServerOptionsExtension
+        {
+            public NoTxSqlServerOptionsExtension()
+            {
+            }
+
+            public NoTxSqlServerOptionsExtension(NoTxSqlServerOptionsExtension copyFrom) : base(copyFrom)
+            {
+            }
+            public override void ApplyServices(IServiceCollection services)
+            {
+                base.ApplyServices(services);
+                services.Remove(services.First((sd) => sd.ServiceType == typeof(ISqlServerConnection)));
+                services.AddScoped<ISqlServerConnection, NoTransactionSqlServerConnection>();
```

Comment:
This is an unusual way to do service replacement. I'm guessing this is because you don't want EF to share the same service provider as the application and you don't want to manage the service provider yourself?

The more normal way to do this would be to change AddDbContext as follows:
```C#
services
    .AddScoped<ISqlServerConnection, NoTransactionSqlServerConnection>()
    .AddEntityFrameworkSqlServer()
    .AddDbContext<ApplicationDbContext>(
        (serviceProvider, options) => options
            .UseInternalServiceProvider(serviceProvider)
            .UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
```

That being said, if you want to do it this way, then its probably fine.